addons/knowledge/controllers/web_unsplash.py

```diff
@@ -0,0 +1,63 @@
+import mimetypes
+import requests
+
+from odoo.addons.web_unsplash.controllers import main
+from odoo import http, tools
+from odoo.http import request
+from odoo.tools.mimetypes import guess_mimetype
+from werkzeug.urls import url_encode
+
+class Web_Unsplash(main.Web_Unsplash):
+
+    @http.route("/web_unsplash/save_random", type='json', auth="user")
+    def save_unsplash_random(self, **kwargs):
+
+        def slugify(s):
+            ''' Keeps only alphanumeric characters, hyphens and spaces from a string.
+                The string will also be truncated to 1024 characters max.
+                :param s: the string to be filtered
+                :return: the sanitized string
+            '''
+            return "".join([c for c in s if c.isalnum() or c in list("- ")])[:1024]
+
+        # Fetch a random image
+        access_key = self._get_access_key()
+        app_id = self.get_unsplash_app_id()
+        if not access_key or not app_id:
+            return {'error': 'key_not_found'}
+        kwargs['client_id'] = access_key
+        req = requests.get('https://api.unsplash.com/photos/random', params=url_encode(kwargs))
+        if req.status_code != requests.codes.ok:
+            return {'error': req.status_code}
+        image_info = req.json()
+
+        # Fetch binary image
+        req = requests.get(image_info['urls']['regular'])
+        if req.status_code != requests.codes.ok:
+            return {'error': req.status_code}
+
+        # Save image in attachments
+        image = tools.image_process(req.content, verify_resolution=True)
+        mimetype = guess_mimetype(image)
+        query = slugify(kwargs.get('query')) + mimetypes.guess_extension(mimetype) or ''
+
+        Attachments = request.env['ir.attachment']
+        res_model = kwargs.get('res_model', 'ir.ui.view')
+        if res_model != 'ir.ui.view' and kwargs.get('res_id'):
+            res_id = int(kwargs['res_id'])
+        else:
+            res_id = None
+        attachment = Attachments.create({
+            'name': '_'.join(['unsplash', image_info['id'], query]),
+            'url': '/' + '/'.join(['unsplash', image_info['id'], query]),
+            'mimetype': mimetype,
+            'raw': image,
+            'public': 'res_model' == 'ir.ui.view',
+            'res_id': res_id,
+            'res_model': res_model,
+            'description': image_info['alt_description'],
+        })
+        attachment.generate_access_token()
+        # Notify unsplash (API requirment)
+        self._notify_download(image_info['links']['download_location'])
```

Comment:
It would be possible. I just had in mind from another task that it's a bad thing to call a method from a http route inside another method, so I tried to avoid doing it. If it's not a problem I will do it

I hesitated to cut and paste "save_unsplash_url" and simply call it when calling the http route, or when we want to save the random image, but I thought that it would be better if I could avoid to modify web_unsplash's controller.